### ISSUE TITLE
Unknown layer type: InnerProduct 

### ISSUE THREAD

**Issue by LilySys** [runhang/caffe-ssd-windows#2]
Hello!
I build caffe_ssd_windows sucessfully, but when run my own prototxt, I got this error:
F0808 19:20:32.686131 12316 layer_factory.cpp:77] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: InnerProduct (known types: AnnotatedData, BatchNorm, Bias, Concat, Convolution, DetectionEvaluate, DetectionOutput, Eltwise, Flatten, Input, LRN, MemoryData, MultiBoxLoss, Permute, Pooling, Power, PriorBox, Python, ReLU, Reshape, Scale, Sigmoid, SmoothL1Loss, Softmax, SoftmaxWithLoss, Split, TanH)
*** Check failure stack trace: ***

**Comment by runhang** [comment]
I fixed this bug at this project,
// Add following function at layer_factory.cpp
// Get normalize layer according to engine.
template 
shared_ptr<Layer > GetNormalizeLayer(const LayerParameter& param) {
return shared_ptr<Layer >(new NormalizeLayer(param));
}
REGISTER_LAYER_CREATOR(Normalize, GetNormalizeLayer);
remove REGISTER_LAYER_CLASS(Normalize);
////////////
// Get InnerProduct layer according to engine.
template <typename Dtype>
shared_ptr<Layer<Dtype> > GetInnerProductLayer(const LayerParameter& param) {
     return shared_ptr<Layer<Dtype> >(new InnerProductLayer(param));
}
REGISTER_LAYER_CREATOR(InnerProduct, GetInnerProductLayer);

**Comment by LilySys** [comment]
REGISTER_LAYER_CLASS() means to register the layer, why you remove it at ***layer.cpp and then re-register at layer_factory?

**Comment by runhang** [comment]
This is a small bug in the Windows environment
